File: Merge-Two-Sorted-Lists.js
1/**
2 * Definition for singly-linked list.
3 * function ListNode(val, next) {
4 *     this.val = (val===undefined ? 0 : val)
5 *     this.next = (next===undefined ? null : next)
6 * }
7 */
8/**
9 * @param {ListNode} list1
10 * @param {ListNode} list2
11 * @return {ListNode}
12 */
13var mergeTwoLists = function(list1, list2) {
14    let list = new ListNode
15    let dummy = list
16    while(list1&&list2){
17        if(list1.val<list2.val){
18            dummy.next = list1
19            list1 = list1.next
20        }else{
21            dummy.next = list2
22            list2 = list2.next
23        }
24        dummy = dummy.next
25    }
26   dummy.next = list1||list2
27    return list.next
28};

File: Merge-Two-Sorted-Lists.cs
1/**
2 * Definition for singly-linked list.
3 * public class ListNode {
4 *     public int val;
5 *     public ListNode next;
6 *     public ListNode(int val=0, ListNode next=null) {
7 *         this.val = val;
8 *         this.next = next;
9 *     }
10 * }
11 */
12public class Solution {
13    public ListNode MergeTwoLists(ListNode l1, ListNode l2) {
14        if(l1 == null) return l2;
15        if(l2 == null) return l1;
16        if(l1.val <= l2.val){
17            l1.next = MergeTwoLists(l1.next,l2);
18            return l1;
19        }else{
20            l2.next = MergeTwoLists(l1,l2.next);
21            return l2;
22        }
23    }
24}